keyboards/rephlex/2k/2k.c

```diff
@@ -0,0 +1,170 @@
+/* Copyright 2023 RephlexZero (@RephlexZero)
+SPDX-License-Identifier: GPL-2.0-or-later */
+#include <stdint.h>
+#include "2k.h"
+#include "quantum.h"
+#include "analog.h"
+#include "eeprom.h"
+#include "config.h"
+
+extern pin_t matrix_pins[MATRIX_ROWS][MATRIX_COLS];
+void         bootmagic_lite(void) {
+    if (analogReadPin(matrix_pins[BOOTMAGIC_LITE_ROW][BOOTMAGIC_LITE_COLUMN]) < 1350) {
+        bootloader_jump();
+    }
+}
+
+#ifdef DEBUG_MATRIX
+static uint8_t i = 0;
+void           housekeeping_task_user(void) {
+    if (i == 0) {
+        uprintf("Mode:%d Actuation Point %d Press/Release sensitivity:%d/%d\n", g_config.mode, g_config.actuation_point, g_config.press_sensitivity, g_config.release_sensitivity);
+        for (uint8_t i = 0; i < MATRIX_ROWS; i++) {
+            uprintf("\n");
+            for (uint8_t j = 0; j < MATRIX_COLS; j++) {
+                uprintf("%d/%d ", keys[i][j].value, analogReadPin(matrix_pins[i][j]));
+            }
+            uprintf("\n");
+        }
+    }
+    i++;
+}
+#endif
+
+analog_config g_config = {
+    .mode                = 1,
+    .actuation_point     = 32,
+    .press_sensitivity   = 32,
+    .release_sensitivity = 32,
+    .press_hysteresis    = 5,
+    .release_hysteresis  = 5
+};
+
+#ifdef VIA_ENABLE
+void values_load(void) {
+    eeconfig_read_kb_datablock(&g_config);
+}
+
+void values_save(void) {
+    eeconfig_update_kb_datablock(&g_config);
+}
+
+void via_init_kb(void) {
+    /* If the EEPROM has the magic, the data is good.
+    OK to load from EEPROM */
+    if (via_eeprom_is_valid()) {
+        values_load();
+    } else {
+        values_save();
+        /* DO NOT set EEPROM valid here, let caller do this */
+    }
+}
+
```

Comment:
Datablock actually has it's own validity check.  No need to use via's.  And by doing so, don't need to use the via specific functions.


```suggestion
void values_load(void) {
    eeconfig_read_kb_datablock(&g_config);
}

void values_save(void) {
    eeconfig_update_kb_datablock(&g_config);
}

void keyboard_post_init_kb(void) {
    values_load();
    keyboard_post_init_kb()
}

#ifdef VIA_ENABLE
```